mobile/src/main/java/org/openhab/habdroid/ui/ViewExtensions.kt

```diff
@@ -111,3 +122,26 @@ fun RemoteViews.duplicate(): RemoteViews {
         clone()
     }
 }
+
+fun MaterialButton.setTextAndIcon(connection: Connection, mapping: LabeledValue) {
+    val iconUrl = mapping.icon?.toUrl(context, true)
+    if (iconUrl == null) {
+        icon = null
+        text = mapping.label
+        return
+    }
+    CoroutineScope(Dispatchers.IO + Job()).launch {
+        val drawable = try {
+            connection.httpClient.get(iconUrl, caching = HttpClient.CachingMode.DEFAULT)
+                .asBitmap(100, 0, ImageConversionPolicy.ForceTargetSize).response
+                .toDrawable(resources)
+        } catch (e: HttpClient.HttpException) {
+            Log.d(WidgetAdapter.TAG, "Error getting icon for button", e)
+            null
+        }
+        withContext(Dispatchers.Main) {
+            icon = drawable
+            text = if (drawable == null) mapping.label else null
```

Comment:
Why hide the label if an icon is given, instead of showing both?

Reply:
I copied the behavior of Basic UI. The label should be used as fallback only. In our case it could be added as content description of the icon.